CHANGELOG.md

```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## 1.8.2
+* Enhancement - Added a functional test for a now fixed TaskEngine deadlock issue.
```

Comment:
I think several of these are not changelog-worthy as they are not user-facing.  In general, things that affect only developers of the agent are reasonable to exclude from the changelog.

I would remove the following:
* Enhancement - Added a functional test for a now fixed TaskEngine deadlock issue.
* Enhancement - Extend the TaskEngine Interface by adding the GetTaskByArn method.
* Enhancement - Improved general stability for integ tests.
* Enhancement - Disabled OOM testing for unsupported kernel configurations.

I would change the duplicate reporting one to read as follows:
* Bug - Fixed an issue where `exec_create` and `exec_start` events were not correctly ignored with some Docker versions.

I would also change the `kill` one to read as follows:
* Bug - Resolved a bug where sending a signal to a container caused the agent to treat the container as dead.